### ISSUE TITLE
build version 0.7 failed: ‘stod’ is not a member of ‘std’ 

### ISSUE THREAD

**Issue by scotthuang1989** [dmlc/xgboost#3158]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

## Environment info
Operating System:
ubuntu 17.10 64bit

Compiler:

gcc (Ubuntu 5.5.0-1ubuntu2) 5.4.1 20171010

`xgboost` version used:

0.7(download from the release page)

## Steps to reproduce

1. cd xgboost-0.7
2. make -j4

# Error: 

> dmlc-core/include/dmlc/./parameter.h: In member function ‘virtual void dmlc::parameter::FieldEntry<float>::Set(void*, const string&) const’:
> dmlc-core/include/dmlc/./parameter.h:991:25: error: ‘stof’ is not a member of ‘std’
>        this->Get(head) = std::stof(value);
>                          ^
> dmlc-core/include/dmlc/./parameter.h: In member function ‘virtual void dmlc::parameter::FieldEntry<double>::Set(void*, const string&) const’:
> dmlc-core/include/dmlc/./parameter.h:1016:25: error: ‘stod’ is not a member of ‘std’

I google this error,  someone said that i need enable c++ 11 compilation. but I don't know what to do. I check the relase note, it said this version support c++ 11. So I think maybe this is bug?

**Comment by khotilov** [comment]
Please follow the instructions from https://xgboost.readthedocs.io/en/latest/build.html
The archives from the release tab do not include the code from the submodules, and you seem to be not telling the whole story of what you did to work around that.

@hcho3 it would perhaps be better if the source code bundles from the releases were self-contained, and also included at least the necessary dmlc-core and rabit submodules code, and maybe nccl and cub as well.

**Comment by scotthuang1989** [comment]
@khotilov 
yes, I did follow this documentation. Here are the 3 commands I use

```
git clone --recursive https://github.com/dmlc/xgboost
cd xgboost/
make -j4
```

I think dmlc-core and rabit already checked out, because I here are last few output of `git clone`:


> Submodule path 'cub': checked out 'b20808b1b04ec3d6a625e51fbc1eb76f337754ad'
> 
> Submodule path 'dmlc-core': checked out 'b5bec5481df86e8e6728d8bd80a61d87ef3b2cd5'
> Submodule path 'nccl': checked out 'faeac8333cec3ed7474dc9f1c35088dcd6eea60b'
> Submodule path 'rabit': checked out 'a764d45cfb438cc9f15cf47ce586c02ff2c65d0f'

Here are output of `gcc --version`

gcc (Ubuntu 5.5.0-1ubuntu2) 5.4.1 20171010

**Comment by khotilov** [comment]
The line numbers in the original error don't seem to match with b5bec54...
Does a test file x.cc containing
```c++
$ cat x.cc
#include <string>
int main(){
     double x = std::stod("123.123");
     return 0;
}
```
compile with `g++ -std=c++11 x.cc`? 
And what was the compiler command that led to the error?

**Comment by hcho3** [comment]
@khotilov The source archives are generated automatically by GitHub. If we upload our own source archive, does it shadow GitHub-generated archive?

**Comment by hcho3** [comment]
Closing for the lack of activity in last 90 days. Any future installation questions should be directed to https://discuss.xgboost.ai/.
